third_party/mkl_dnn/mkldnn_acl.BUILD

```diff
@@ -44,10 +45,10 @@ _DNNL_RUNTIME_OMP = {
     "#cmakedefine01 BUILD_PRIMITIVE_GPU_ISA_ALL": "#define BUILD_PRIMITIVE_GPU_ISA_ALL 0",
     "#cmakedefine01 BUILD_GEN9": "#define BUILD_GEN9 0",
     "#cmakedefine01 BUILD_GEN11": "#define BUILD_GEN11 0",
-    "#cmakedefine01 BUILD_XELP": "#define BUILD_XELP 0",
-    "#cmakedefine01 BUILD_XEHP": "#define BUILD_XEHP 0",
     "#cmakedefine01 BUILD_XEHPG": "#define BUILD_XEHPG 0",
     "#cmakedefine01 BUILD_XEHPC": "#define BUILD_XEHPC 0",
+    "#cmakedefine01 BUILD_XELP": "#define BUILD_XELP 0",
+    "#cmakedefine01 BUILD_XEHP": "#define BUILD_XEHP 0",
```

Comment:
It seems to work in the x86 PR (https://github.com/tensorflow/tensorflow/pull/55354) though?

Reply:
The order also works before the version update. Maybe there was a recent change in Bazel and newer Bazel version handles the replaces differently. (And the x86 PR might be tested with older Bazel version.) What Bazel version are you using?

@sachinmuradi What Bazel version are you using?

Reply:
@penpornk I am using bazel 5.0.0. Also, previously we had similar issue for oneDNN 2.5.1, when updating oneDNN from 2.5-rc to 2.5.1 we had to change order of #cmakedefine01 BUILD_XEHP by putting it after #cmakedefine01 BUILD_XEHPC": "#define BUILD_XEHPC 0".

Reply:
@sachinmuradi Thank you for the quick reply! It's weird that Bazel's matching behavior seems to be changing back and forth (either that or it's somewhat non-deterministic).

@nSircombe Let's keep `XEHPC` first in this PR. :)

Reply:
I've tested this patch building master with Bazel 5.0.0.
But I originally encountered, and resolved, the issue with the ordering of the `cmakedefine`s on a build with Bazel 4.2.2.
So I don't think I've actually tested the original order with Bazel 5.0.0. 

I'll give it a go.

Reply:
I think the new order is safer anyway. Maybe update the PR that longer matches should be put first just for safety?

Reply:
> Maybe add a comment in the code that longer matches should be put first just for safety?

OK, I'll add something.

Reply:
@penpornk, @sachinmuradi  - I've added a note, is that what you had in mind?

...I'm still puzzled by this error, and why it's showing up in our builds. I have a hunch that the root cause is present in the x86 builds (i.e. the mis substitution), but that it isn't resulting in a compile time error, possibly due to the toolchain being used. It might be prudent to re-order `mkldnnl_v1.BUILD` to use the same order as `mkldnn_acl.BUILD`?

Reply:
I agree that it's better to stay safe. I'll request changes in the other PR too.

Reply:
Just realized that `BUILD_XEHP` is already last in the x86 substitution list. I misread it earlier. Sorry about that! (So nothing needs to be done for the x86 PR.)

Reply:
Ah, yes! Didn't spot that
I'll update this to remove that comment and match the order in `mkldnn_v1.BUILD`

Reply:
fixed with the latest version.